Kudu.Services/LinuxConsumptionInstanceAdmin/EnvironmentReadyCheckMiddleware.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Threading.Tasks;
+using Kudu.Contracts;
+using Microsoft.AspNetCore.Http;
+
+namespace Kudu.Services.LinuxConsumptionInstanceAdmin
+{
+    /// <summary>
+    /// Middleware used in Linux consumption to delay requests until specialization is complete.
+    /// </summary>
+    public class EnvironmentReadyCheckMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public EnvironmentReadyCheckMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext httpContext, ILinuxConsumptionEnvironment environment)
+        {
+            if (environment.DelayRequestsEnabled)
+            {
+                await environment.DelayCompletionTask;
```

Comment:
This is to make sure specialization is complete before we start processing any requests.
Until now specialization was very light weight and doesn't take long to complete but since we are mounting the file share now it could take longer so this check will avoid the race condition.

The implementation was there already but it was being hidden by an override for DelayCompletionTask which I removed in this PR.

Reply:
Are specializations way of executing container operations without a recycle ?

Reply:
Correct. It is a way to inject additional settings and content without recycling the whole container. 

The containers start up differently in consumption vs  Appservice plan / premium skus. In consumption the containers start in placeholder mode with no customer content so we need a way to inject these settings at a later time.